### ISSUE TITLE
Gateway routing issues

### ISSUE THREAD

**Issue by Sidefix** [sorz/sstp-server#39]
Hello,

I have managed to set up the server and run it locally, however I have a question in regards to expected functionality and configuration.

The configuration specifies that the server can self-assign itself an IP address. The default value is 192.168.20.1, however this is configurable. However, it is not specified in the instructions if a specific networking adapter needs to be set up to facilitate this.

To be more specific, when running ifconfig I do not have any IP address that matches the one configured for the server. However, I was able to successfully connect to the server with Windows machines. The issue now becomes that these machines are **not able** to connect to the internet while the VPN connection is up; the issue is mitigated if I disable the server-inherited default gateway on the client.

To summarize, I cannot access the internet from clients connected to the server and I can't seem to figure out how to set up gateway routing appropriately to facilitate this because I cannot identify an adapter on the server that I can work with.

The server setup instructions do not specify anything in this regard, so I am either missing something that is implicitly known for server administrators, or I did something wrong, or there is another issue.

Thank you in advance for your help.

**Comment by sorz** [comment]
Hi,

Once a client connected to the server, a new network interface, usually called `ppp0` or something, will be created on the server. The server-side IP address (default to 192.168.20.1) will be assigned to that ppp interface. It is done automatically, no other work need.

In other to let the client accessing Internet via server, you need to turn on IP forwarding and configure firewall (usually iptables) to allow forwarding and probably doing NAT. There are lots of PPTP setup guides on Internet you may refer to, such as https://wiki.archlinux.org/index.php/PPTP_server. PPTP's is roughly the same as SSTP's on this part.

**Comment by Sidefix** [comment]
@sorz I tore my hair out trying to get this to work, it never occurred to me to follow the guides for PPTP servers.
I have followed all the steps from that article and configured the server appropriately and it now works perfectly.

I just wanted to offer my heartfelt thanks for your quick and informative answer :)

Cheers
